server/src/main/java/org/eclipse/hono/application/HonoApplication.java

```diff
@@ -60,61 +59,98 @@
     }
 
+    public HonoConfigProperties getHonoConfig() {
+        return honoConfig;
+    }
+
     /**
      * @param vertx the vertx to set
      */
     @Autowired
-    public void setVertx(final Vertx vertx) {
+    public final void setVertx(final Vertx vertx) {
         this.vertx = vertx;
     }
 
+    public Vertx getVertx() {
+        return vertx;
+    }
+
     /**
      * @param registrationService the registrationService to set
      */
     @Autowired
-    public void setRegistrationService(final RegistrationService registrationService) {
+    public final void setRegistrationService(final RegistrationService registrationService) {
         this.registrationService = registrationService;
     }
 
+    public RegistrationService getRegistrationService() {
+        return registrationService;
+    }
+
     /**
      * @param authenticationService the authenticationService to set
      */
     @Autowired
-    public void setAuthenticationService(final AuthenticationService authenticationService) {
+    public final void setAuthenticationService(final AuthenticationService authenticationService) {
         this.authenticationService = authenticationService;
     }
 
+    public AuthenticationService getAuthenticationService() {
+        return authenticationService;
+    }
+
     /**
      * @param authorizationService the authorizationService to set
      */
     @Autowired
-    public void setAuthorizationService(final AuthorizationService authorizationService) {
+    public final void setAuthorizationService(final AuthorizationService authorizationService) {
         this.authorizationService = authorizationService;
     }
 
+    public AuthorizationService getAuthorizationService() {
+        return authorizationService;
+    }
+
     /**
      * @param serverFactory the serverFactory to set
      */
     @Autowired
-    public void setServerFactory(final HonoServerFactory serverFactory) {
+    public final void setServerFactory(final HonoServerFactory serverFactory) {
         this.serverFactory = serverFactory;
     }
 
+    public HonoServerFactory getServerFactory() {
+        return serverFactory;
+    }
+
     /**
-     * Deploys all verticles the Hono server consists of.
+     * Starts the Hono server.
      */
     @PostConstruct
-    public void registerVerticles() {
+    public final void startup() {
+        boolean startupSuccessful = registerVerticles();
+        if (startupSuccessful) {
+            postRegisterVerticles();
+        }
+    }
+    
+    /**
+     * Deploys all verticles the Hono server consists of.
+     * 
+     * @return true if deployment was successful
+     */
+    private boolean registerVerticles() {
 
         if (vertx == null) {
             throw new IllegalStateException("no Vert.x instance has been configured");
         }
 
         final CountDownLatch startupLatch = new CountDownLatch(1);
+        final AtomicBoolean startupFailed = new AtomicBoolean(false);
```

Comment:
I guess the code would be easier to read if the boolean would indicate success, i.e. `startupSucceeded`, don't you think?